### PR TITLE
Add cli support for CRUD mutators

### LINKED ISSUE #1119: Add cli management for mutators
Update sensuctl with the ability to CRUD mutators.

### PR DESCRIPTION
**Pull request by mercul3s**
## What is this change?

Adds support for mutators to the cli.

## Why is this change necessary?

Closes #1119 

## Does your change need a Changelog entry?

Yup!

## Do you need clarification on anything?

Nope!

## Were there any complications while making this change?

I also noticed an incorrect flag for filters in the handler interactive code, so I fixed it.

### CODE REVIEW

File: cli/commands/handler/interactive.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func (opts *handlerOpts) withHandler(handler *types.Handler) {
 
 func (opts *handlerOpts) withFlags(flags *pflag.FlagSet) {
 	opts.Command, _ = flags.GetString("command")
-	opts.Filters, _ = flags.GetString("handlers")
+	opts.Filters, _ = flags.GetString("filters")
```
Comment (jamesdphillips) [inline cli/commands/handler/interactive.go]: 😰

File: cli/commands/mutator/info.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+			if format == "json" {
+				if err := helpers.PrintJSON(r, cmd.OutOrStdout()); err != nil {
+					return err
+				}
+			} else {
+				printToList(r, cmd.OutOrStdout())
+			}
+
+			return nil
```
Comment (jamesdphillips) [inline cli/commands/mutator/info.go]: Maybe we could simplify to

```golang
if format == "json" {
  return helpers.PrintJSON(r, cmd.OutOrStdout())
}
printToList(r, cmd.OutOrStdout())
return nil
```

Comment (jamesdphillips) [reply cli/commands/mutator/info.go]: (On a related note.. honestly `printToList` should probably return an error, but, that's an issue with my `list` package and totally out of scope.)

File: cli/commands/handler/interactive.go [hunk 3]
```diff
@@ -54,7 +54,7 @@ func (opts *handlerOpts) withHandler(handler *types.Handler) {
 
 func (opts *handlerOpts) withFlags(flags *pflag.FlagSet) {
 	opts.Command, _ = flags.GetString("command")
-	opts.Filters, _ = flags.GetString("handlers")
```
Comment (palourde) [inline cli/commands/handler/interactive.go]: giphy

File: cli/commands/mutator/update.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			}
+
+			// Administer questionnaire
+			opts := newMutatorOpts()
+			opts.withMutator(mutator)
+			if err := opts.administerQuestionnaire(true); err != nil {
+				return err
+			}
+
+			// Apply given arguments to check
```
Comment (palourde) [inline cli/commands/mutator/update.go]: Maybe just update this comment 😄

File: cli/client/mutator.go [hunk 5]
```diff
@@ -25,14 +25,71 @@ func (client *RestClient) ListMutators() ([]types.Mutator, error) {
...
+		return err
+	}
+
+	res, err := client.R().SetBody(bytes).Patch("/mutators/" + url.PathEscape(mutator.Name))
+	if err != nil {
+		return err
+	}
+
+	if res.StatusCode() >= 400 {
+		return fmt.Errorf("%v", res.String())
```
Comment (jamesdphillips) [inline cli/client/mutator.go]: Unfortunately it's used inconsistently but ideally you could use `unmarshalError`. See `client/check.go` for example https://github.com/sensu/sensu-go/blob/master/cli/client/check.go#L54-L56.

File: cli/commands/mutator/info.go (hunk 2 above)
Comment (mercul3s) [reply cli/commands/mutator/info.go]: oh yeah, that is much cleaner. Thanks for the suggestion, I'll change it!

Comment (mercul3s) [reply cli/commands/mutator/info.go]: Should I add an issue for printToList returning an error?

Comment (jamesdphillips) [reply cli/commands/mutator/info.go]: No worries, I can do it.

**Comment by mercul3s** [comment]
One of the tests is oddly failing in the linter stage from OOM errors with mega check; everything else passes, and the linter is fine locally so I'm going to go ahead and merge this.

**Pull request merged**